RAD-5646-Monitor-and-Alert-on-Event-Counts-via-Scheduled-Task

josecuadrarad opened the pull request:
This script checks if the number of events, by type, in the RECENT_PERIOD exceeds the AVERAGE_EVENTS_PER_PERIOD, by more than the TYPE_THRESHOLD percentage, set the alarm data point for this type to 1 (Active alarm). Else, set the data point to 0 (Inactive alarm).

josecuadrarad committed:
RAD-5646 Monitor and Alert on Event Counts via Scheduled Task

josecuadrarad committed:
RAD-5646 Monitor and Alert on Event Counts via Scheduled Task

josecuadrarad committed:
RAD-5646 Monitor and Alert on Event Counts via Scheduled Task

josecuadrarad committed:
RAD-5646 Monitor and Alert on Event Counts via Scheduled Task

jpuccinit commented on event-count-monitoring.js:
```diff
@@ -0,0 +1,158 @@
+
+
+console.log("=============================== EXCEEDS THRESHOLD? =============================== ")
+var exceedsThresholdIndex = 0;
```
please use const instead of var

jazdw replied:
Use `let` for variables, use `const` for constants.

jpuccinit reviewed (commented):
Please change var to  let.

jazdw reviewed (approved):
Thanks @josecuadrarad

josecuadrarad committed:
RAD-5646 Monitor and Alert on Event Counts via Scheduled Task

aasbra committed:
Adding data point type support

• Error logging when an alarm data point does not exist in Mango
• Validation to ensure the number of event types and alarm data point XIDs are the same
• Moved everything inside a function so execution will stop when an error is encountered
• Support for Binary, Numeric, and Multistate alarm data point types
• Support for getDataType and getDataTypeId methods, ensuring backwards compatibility with earlier version of Mango 4
